src/tasks/DockerPseudoShell.ts

```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import { CancellationToken, CancellationTokenSource, Event, EventEmitter, Pseudoterminal, TerminalDimensions, WorkspaceFolder } from 'vscode';
+import { CommandLineBuilder } from '../utils/commandLineBuilder';
+import { resolveVariables } from '../utils/resolveVariables';
+import { DockerBuildTask } from './DockerBuildTaskProvider';
+import { DockerRunTask } from './DockerRunTaskProvider';
+import { DockerTaskProviderBase } from './DockerTaskProviderBase';
+import { DockerTaskExecutionContext } from './TaskHelper';
+
+const DEFAULT = '0m';
+const DEFAULTBOLD = '0;1m';
+const RED = '31m';
+const YELLOW = '33m';
+
+export class DockerPseudoShell implements Pseudoterminal {

```

Comment:
Nit: maybe make naming consistent with `DockerPseudoterminal`?

Reply:
Yeah, can do that. The reason I named it `DockerPseudoShell` is that we're using it more like a simple shell than a full-blown `Pseudoterminal`.